lib/phoenix/transports/websocket.ex

```diff
@@ -1,5 +1,9 @@
 defmodule Phoenix.Transports.WebSocket do
   @moduledoc false
+  @behaviour Plug
```

Comment:
This is now a Plug that gets routed to from the websocket routes within `socket` declarations in Endpoint. It uses the `Plug.Conn.upgrade_adapter/3` function to signal upgrades back to the HTTP adapter.